api/server/server.go

```diff
@@ -167,6 +169,11 @@ func NewFromEnv(ctx context.Context, opts ...ServerOption) *Server {
 	opts = append(opts, WithNodeCert(getEnv(EnvCert, "")))
 	opts = append(opts, WithNodeCertKey(getEnv(EnvCertKey, "")))
 	opts = append(opts, WithNodeCertAuthority(getEnv(EnvCertAuth, "")))
+	ridProvider := &RIDProvider{
+		HeaderName:   getEnv(EnvRidHeader, "fn-request-id"),
+		RIDGenerator: common.FnRequestID,
```

Comment:
could inline this one even: `RIDGenerator: func() string { return id.New().String() }` - pretty terse, one less exposed method from common

Reply:
nvm, not as easy the way it's set up now. i'd think we'd want the same `if ridFound { return rid } else { return gen() }` logic for any rid provider, though, and maybe `RIDProvider` should only be a generator as opposed to a branch to handle a request id existing. I don't have full context here - if that's what we need, then that is perfectly ok.

Reply:
I see it more as a generator so I'd prefer to leave it as it is, it gives us more flexibility